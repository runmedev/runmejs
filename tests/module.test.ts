import fs from 'node:fs/promises'
import url from 'node:url'
import path from 'node:path'
import { test, expect, beforeAll } from 'vitest'

import { run, createServer } from '../src/index.js'

const __dirname = url.fileURLToPath(new URL('.', import.meta.url))

// reliable snapshots x-platform and node versions
function toJsonSnapshot(obj: any) {
  return JSON.stringify(obj, null, 1)
}

beforeAll(async () => {
  await fs.rm(
    path.resolve(__dirname, '..', '.bin'),
    { recursive: true, force: true }
  )
})

test('run', async () => {
  expect(toJsonSnapshot(await run(['helloWorld'])))
    .toMatchSnapshot()
  expect(toJsonSnapshot(await run(['fail'], { ignoreReturnCode: true })))
    .toMatchSnapshot()
})

test('createServer', async () => {
  const server = await createServer()
  expect(toJsonSnapshot(await run(['export', 'print'], { server })))
    .toMatchSnapshot()
  server.kill()
})
