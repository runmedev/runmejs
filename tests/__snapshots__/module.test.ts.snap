// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`createServer 1`] = `
"{
 \\"exitCode\\": 0,
 \\"stdout\\": \\" ►  Running task export...\\\\n ►  ✓ Task export exited with code 0\\\\n ►  Running task print...\\\\nexported FOO=bar\\\\r\\\\n ►  ✓ Task print exited with code 0\\\\n\\",
 \\"stderr\\": \\"\\"
}"
`;

exports[`run 1`] = `
"{
 \\"exitCode\\": 0,
 \\"stdout\\": \\"Hello World\\\\r\\\\n\\",
 \\"stderr\\": \\"\\"
}"
`;

exports[`run 2`] = `
"{
 \\"exitCode\\": 1,
 \\"stdout\\": \\"failed to run command \\\\\\"fail\\\\\\": exit code: 1\\\\n\\",
 \\"stderr\\": \\"\\"
}"
`;
